consensus/istanbul/core/prepare_test.go

```diff
@@ -28,6 +28,62 @@ import (
 	"github.com/ethereum/go-ethereum/crypto/bls"
 )
 
+func TestHandlePreparedCertificate(t *testing.T) {
+	N := uint64(4) // replica 0 is the proposer, it will send messages to others
+	F := uint64(1)
+	sys := NewTestSystemWithBackend(N, F)
+	view := istanbul.View{
+		Round:    big.NewInt(0),
+		Sequence: big.NewInt(1),
+	}
+	proposal := makeBlock(0)
+
+	testCases := []struct {
```

Comment:
Are there any other error cases we're missing?

Reply:
Missing:
* `errInvalidPreparedCertificateProposal`
* `errInvalidPreparedCertificateMsgSignature`
* Plus errors directly return from `message.PayloadNoSig`, `c.validateFn`, and `c.verifyCommittedSeal`